test/metabase/sync/util_test.clj

```diff
@@ -188,3 +189,40 @@
         (testing "has-step-duration?"
           (is (= true
                  (str/includes? results "4.0 s"))))))))
+
+(deftest error-handling-test
+  (testing "A ConnectException will cause sync to stop"
+    (mt/dataset sample-dataset
```

Comment:
Indentation is funky here 😬

You don't need to use the sample dataset if you're not testing anything in it (from what I can tell it doesn't seem like you are). mt/db and mt/id and the like will just use the test-data dataset by default if you don't specify anything different. So you could remove this form entirely

Reply:
Triggering the sync flow seemed to require a proper database, I got an error trying to do it any other way.

Reply:
It might have been because a `Database` was saved in your app DB from a previous session. Since the H2 DBs are in-memory they stop working if you restart your REPL. It's supposed to detect those situations and recreate them if that happens. So you'd probably run in to this too if you quit the REPL and restarted and tried to run this test again

```clj
(toucan.db/delete! 'Database :engine "h2", :name "test-data")
```

should probably fix this not working from the REPL